### PR TITLE
add test with fields=:all in getOrgUnits

### PR DESCRIPTION
**Pull request by sam-bao**
@maxwellchandler 

Found an issue with an edge case for getOrgUnit Groups. It is possible to specify `fields=:all` as a special case and get back all fields. I am getting an error when I try this:

```
getOrgUnitGroups(
+     c("CHC", "Country"), 
+     by = shortName, 
+     fields = ":all",
+     base_url = "https://play.dhis2.org/2.34/"
+   )
 Hide Traceback
 
 Rerun with Debug
 Error in `[.data.frame`(data, , name_reduce) : undefined columns selected 
4.
stop("undefined columns selected") 
3.
`[.data.frame`(data, , name_reduce) at getOrgUnitGroups.R#88
2.
data[, name_reduce] at getOrgUnitGroups.R#88
1.
getOrgUnitGroups(c("CHC", "Country"), by = shortName, fields = ":all", 
    base_url = "https://play.dhis2.org/2.34/") 
```

Note, getMetadata works fine

```
getMetadata(organisationUnitGroups, shortName %din% c("CHC", "Country"), fields = ":all")
     code             lastUpdated          id
1     CHC 2014-09-29T20:28:01.093 CXw2yu5fodb
2 Country 2011-12-24T12:24:24.017 RpbiCJpIYEj
                                                                  href                 created    name
1 https://play.dhis2.org/2.34.0/api/organisationUnitGroups/CXw2yu5fodb 2012-11-13T14:35:22.574     CHC
2 https://play.dhis2.org/2.34.0/api/organisationUnitGroups/RpbiCJpIYEj 2011-12-24T12:24:24.017 Country
  shortName symbol displayName publicAccess displayShortName externalAccess dimensionItem favorite
1       CHC 16.png         CHC     rw------              CHC          FALSE   CXw2yu5fodb    FALSE
2   Country   <NA>     Country     rw------          Country          FALSE   RpbiCJpIYEj    FALSE
        dimensionItemType favorites translations
```

Additional details from the DHIS2 Docs:
https://docs.dhis2.org/master/en/developer/html/dhis2_developer_manual_full.html#webapi_metadata_field_filter
image

### CODE REVIEW

**Comment by sam-bao** [comment]
This makes knowing if the `by` property is included in `fields` much more complicated. It is not immediately obvious. I think maybe we block the use of any preset except for "all" and "*", and if fields = all or * we skip name_reduce?

**Comment by maxwellchandler** [comment]
@sam-bao  I added some logic to take care of this, whats weird is that calling the api with :all,shortName,name returns 29 rows while calling with just :all returns 30. So I couldnt just do a try-catch on the subset because the data returned is already different.

File: R/getOrgUnitGroups.R [hunk 1]
```diff
@@ -33,13 +33,17 @@ getOrgUnitGroups <- function(values,
...
   name_reduce <- NULL
+  name_reduce_skip <- F
 
   # by can come in as string or NSE, convert to string
   by <- try(as.character(rlang::ensym(by)), silent = TRUE)
 
   default_fields <- if (is.null(fields)) {
     c(by, "name", "id")
-  } else if (!("name" %in% fields) && !(any(grepl("name", fields)))) {
+  } else if(any(grepl("all|//*|identifiable|nameable|persisted|owner", fields))){
```
Comment (sam-bao) [inline R/getOrgUnitGroups.R:43]: We should match on :all, :identifiable, ... as they would appear in an api call as `fields=:all` and `fields=:identifiable`.

Please add comments to your code to explain the why of your logic. In this case something like. 
> Checking for metadata field filter presets that include DHIS2 identifiable properties. If included we can skip name reduce.

**Comment by sam-bao** [comment]
@maxwellchandler I am not able to replicate your issue with getting a different number of columns. It could be that the underlying data were changing while you were testing the API calls out:

```
> datimutils::getMetadata(organisationUnitGroups, shortName %din% c("CHC", "Country"), fields = ":all") %>% NCOL()
[1] 31
> datimutils::getMetadata(organisationUnitGroups, shortName %din% c("CHC", "Country"), fields = ":all,shortName") %>% NCOL()
[1] 31
> datimutils::getMetadata(organisationUnitGroups, shortName %din% c("CHC", "Country"), fields = ":all,shortName,name") %>% NCOL()
[1] 31
```

The potential for underlying data to change is one of the reasons we create the mocks.
